src/p4est_io.h

```diff
@@ -22,13 +22,53 @@
   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
 */
 
+/** \file p4est_io.h
+ *
+ * Provide functions to serialize/deserialize a forest.
+ * Some are used as building blocks for \ref p4est_load and \ref p4est_save.
+ * Others allow for saving and loading user-defined data to a parallel file.
+ *
+ * Furthermore, this module provides functions to write and read general data
+ * files associated with a p4est.
+ */
+
 #ifndef P4EST_IO_H
 #define P4EST_IO_H
 
 #include <p4est.h>
 
 SC_EXTERN_C_BEGIN;
 
+#define P4EST_DATA_FILE_EXT "p4data" /**< file extension of p4est data files */
+#define P4EST_MAGIC_NUMBER "p4data0" /**< magic string for p4est data files */
+#define P4EST_NUM_METADATA_BYTES 96 /**< number of file metadata bytes */
+#define P4EST_NUM_ARRAY_METADATA_BYTES 14 /**< number of array metadata bytes */
+/* subtract 2 for '\n' at the beginning and end of the array metadata */
+#define P4EST_NUM_ARRAY_METADATA_CHARS (P4EST_NUM_ARRAY_METADATA_BYTES - 2) /**< number of array metadata chars */
+#define P4EST_BYTE_DIV 16 /**< All data blocks are padded to be divisible by this. */
+#define P4EST_MAX_NUM_PAD_BYTES (P4EST_BYTE_DIV + 1) /**< We enforce to pad in any
+                                                          case and the padding string
+                                                          needs to contain two
+                                                          newline characters and
+                                                          therefore this is the
+                                                          maximal number of pad
+                                                          bytes. */
+#define P4EST_NUM_USER_STRING_BYTES 48 /**< number of user string bytes */
+#define P4EST_NUM_FIELD_HEADER_BYTES (2 + P4EST_NUM_ARRAY_METADATA_BYTES + P4EST_NUM_USER_STRING_BYTES)
+                                     /**< number of bytes of one field header */
+#define P4EST_FILE_COUNT_ERROR -1 /**< All other error codes are defined by MPI or are
+                                     errno. This error code is used to indicate a read
```

Comment:
The reference to errno may now be outdated (everywhere in p4est).

Reply:
I changed this comment and checked the rest of the documentation.